core/genesis.go

```diff
@@ -273,7 +277,7 @@ func SetupGenesisBlockWithOverride(db ethdb.Database, genesis *Genesis, override
 			genesis = DefaultGenesisBlock()
 		}
 		// Ensure the stored genesis matches with the given one.
-		hash := genesis.ToBlock(nil).Hash()
+		hash := genesis.ToBlock().Hash()
```

Comment:
and maybe split this in two functions:

```golang
genesis.flush() // store the data so that ToBlock().Root is valid
hash := genesis.ToBlock().Hash() // Here `ToBlock` doesn't write anything on disk.
```

Reply:
Actually in `ToBlock`, `genesis.flush` is used to compute the state root. All the flushed states will be discarded since they are in a memorydb.